client/my-sites/stats/stats-module/index.jsx

```diff
@@ -292,35 +287,6 @@ class StatsModule extends Component {
 		const summaryLink = ! this.props.hideSummaryLink && this.getSummaryLink();
 		const displaySummaryLink = data && summaryLink;
 		const isAllTime = this.isAllTimeList();
-		const isStatsNavigationImprovementEnabled = config.isEnabled( 'stats/navigation-improvement' );
-
-		const renderDownloadCsv = () => {
-			// Disable the Download button for the new navigation.
-			if ( isStatsNavigationImprovementEnabled ) {
-				return null;
-			}
-
-			// Disable for the email module as it doesn't work correctly.
-			if ( statType === 'statsEmailsSummary' ) {
-				return null;
-			}
-
-			if ( gateDownloads ) {
-				return <DownloadCsvUpsell siteId={ siteId } borderless />;
-			}
-
-			return (
-				<DownloadCsv
-					statType={ statType }
-					query={ query }
-					path={ path }
-					period={ period }
-					skipQuery={ skipQuery }
-				/>
-			);
-		};
-
-		const downloadCsv = renderDownloadCsv();
```

Comment:
It always returns `null`, so it is no longer necessary to be passed to the `StatsListCard` below.